catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPostProcessPlugin.java

```diff
@@ -0,0 +1,367 @@
+
+            if (!isNitfMimeType(processResource.getMimeType())) {
+                LOGGER.trace("skipping process item: filename={} mimeType={}",
+                        processResource.getFilename(),
+                        processResource.getMimeType());
+            } else {
+                try {
+                    BufferedImage renderedImage = renderImage(processResource);
+
+                    if (renderedImage != null) {
+                        // thumbnail
+                        addThumbnailToMetacard(metacard, renderedImage);
+
+                        // overview
+                        try {
+                            ProcessResource overviewImageProcessResource = createDerivedImage(
+                                    processResourceItem.getMetacard()
+                                            .getId(),
+                                    OVERVIEW,
+                                    renderedImage,
+                                    metacard,
+                                    calculateOverviewWidth(renderedImage),
+                                    calculateOverviewHeight(renderedImage));
+
+                            T overviewProcessResourceItem = createResourceItem(processResourceItem,
+                                    overviewImageProcessResource);
+                            newProcessResourceItems.add(overviewProcessResourceItem);
+                        } catch (IOException e) {
+                            LOGGER.debug("Unable to generate Overview image from nitf. {}.",
+                                    e.getMessage());
+                        }
+
+                        // original
+                        try {
+                            ProcessResource originalImageProcessResource = createDerivedImage(
+                                    processResourceItem.getMetacard()
+                                            .getId(),
+                                    ORIGINAL,
+                                    renderedImage,
+                                    metacard,
+                                    renderedImage.getWidth(),
+                                    renderedImage.getHeight());
+
+                            T originalProcessResourceItem = createResourceItem(processResourceItem,
+                                    originalImageProcessResource);
+                            newProcessResourceItems.add(originalProcessResourceItem);
+                        } catch (IOException e) {
+                            LOGGER.debug("Unable to generate Original image from nitf. {}.",
+                                    e.getMessage());
+                        }
+                    }
+                } catch (IOException | ParseException | NitfFormatException e) {
+                    LOGGER.debug("Unable to render image from nitf. {}.", e.getMessage());
+                }
+            }
+        }
+
+        processResourceItems.addAll(newProcessResourceItems);
```

Comment:
I refactored to combine this method with the method before so that the input argument is not modified.